training/creating_dataset.py

```diff
@@ -0,0 +1,125 @@
+import os.path
+import traceback
+from copy import deepcopy
+from typing import List
+
+import pandas as pd
+
+from training.analysis_functions import (
+    import_merged_bestiaries,
+    unpack_column,
+    DataFrameType,
+)
+
+
+def create_dataframe(
+    books: List[str] = [
+        "../pathfinder_2e_data/pathfinder-bestiary.db",
+        "../pathfinder_2e_data/pathfinder-bestiary-2.db",
+        "../pathfinder_2e_data/pathfinder-bestiary-3.db",
+    ],
+    characteristics: List[str] = [
+        "system/abilities",
+        "system/attributes/ac",
+        "system/attributes/hp",
+    ],
+) -> DataFrameType:
+    """
+
+    :param books: list of paths of books to load
+    :param characteristics: list of characteristics to load
+    :return: DataFrame with monsters(nps) form chosen books and with chosen characteristics and their origin book
+    """
+    loading_methods = {
+        "system/abilities": move_values_level_up(value_name="mod"),
+        "system/attributes/ac": load_sub_column_as_value("ac"),
+        "system/attributes/hp": load_sub_column_as_value("hp"),
+    }
+    try:
+        check_paths(books)
+    except:
+        traceback.print_exc()
+
+    bestiary = import_merged_bestiaries(books)
+    bestiary = bestiary[bestiary["type"] == "npc"]
+    df = create_df_with_basic_values(bestiary)
+
+    for characteristic in characteristics:
+        if (loading_method := loading_methods.get(characteristic)) is not None:
+            part = loading_method(get_sub_column(bestiary, characteristic))
+            df = pd.concat(
+                [
+                    df,
+                    part,
+                ],
+                axis=1,
+            )
+
+    return df
+
+
+def check_paths(paths: List[str]) -> None:
```

Comment:
Tbh nie wiem, czy rzucanie wyjątkami to nie jest przerost formy nad treścią, może lepiej zamienić tą funkcję na is_path_correct(), która będzie zwracała True jak ścieżka jest git i False wpp